### ISSUE TITLE
KeyError: 'reference'

### ISSUE THREAD

**Issue by SilasK** [MrOlm/drep#120]
We are using drep v3.2 and get the following error.

***************************************************
    ..:: dRep dereplicate Step 1. Filter ::..
***************************************************
    
Will filter the genome list
5 genomes were input to dRep
Calculating genome info of genomes
100.00% of genomes passed length filtering
100.00% of genomes passed checkM filtering
***************************************************
    ..:: dRep dereplicate Step 2. Cluster ::..
***************************************************
    
Running primary clustering
Running pair-wise MASH clustering
3 primary clusters made
Running secondary clustering
3. Nevermind! Skipping secondary clustering
Step 4. Return output
***************************************************
    ..:: dRep dereplicate Step 3. Choose ::..
***************************************************
    
Loading work directory
Traceback (most recent call last):
  File "/scratch/ref/gdlab/metagenome-atlas_db/2.6a1/conda_envs/85ea1ba518acc8559abd39ebd8927742/lib/python3.6/site-packages/pandas/core/indexes/base.py", line 2898, in get_loc
    return self._engine.get_loc(casted_key)
  File "pandas/_libs/index.pyx", line 70, in pandas._libs.index.IndexEngine.get_loc
  File "pandas/_libs/index.pyx", line 101, in pandas._libs.index.IndexEngine.get_loc
  File "pandas/_libs/hashtable_class_helper.pxi", line 1675, in pandas._libs.hashtable.PyObjectHashTable.get_item
  File "pandas/_libs/hashtable_class_helper.pxi", line 1683, in pandas._libs.hashtable.PyObjectHashTable.get_item
KeyError: 'reference'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/scratch/ref/gdlab/metagenome-atlas_db/2.6a1/conda_envs/85ea1ba518acc8559abd39ebd8927742/bin/dRep", line 32, in <module>
    Controller().parseArguments(args)
  File "/scratch/ref/gdlab/metagenome-atlas_db/2.6a1/conda_envs/85ea1ba518acc8559abd39ebd8927742/lib/python3.6/site-packages/drep/controller.py", line 100, in parseArguments
    self.dereplicate_operation(**vars(args))
  File "/scratch/ref/gdlab/metagenome-atlas_db/2.6a1/conda_envs/85ea1ba518acc8559abd39ebd8927742/lib/python3.6/site-packages/drep/controller.py", line 48, in dereplicate_operation
    drep.d_workflows.dereplicate_wrapper(kwargs['work_directory'],**kwargs)
  File "/scratch/ref/gdlab/metagenome-atlas_db/2.6a1/conda_envs/85ea1ba518acc8559abd39ebd8927742/lib/python3.6/site-packages/drep/d_workflows.py", line 45, in dereplicate_wrapper
    drep.d_choose.d_choose_wrapper(wd, **kwargs)
  File "/scratch/ref/gdlab/metagenome-atlas_db/2.6a1/conda_envs/85ea1ba518acc8559abd39ebd8927742/lib/python3.6/site-packages/drep/d_choose.py", line 72, in d_choose_wrapper
    Gdb = add_centrality(wd, Gdb, **kwargs)
  File "/scratch/ref/gdlab/metagenome-atlas_db/2.6a1/conda_envs/85ea1ba518acc8559abd39ebd8927742/lib/python3.6/site-packages/drep/d_choose.py", line 306, in add_centrality
    Ndb['cluster_1'] = Ndb['reference'].map(g2c)
  File "/scratch/ref/gdlab/metagenome-atlas_db/2.6a1/conda_envs/85ea1ba518acc8559abd39ebd8927742/lib/python3.6/site-packages/pandas/core/frame.py", line 2906, in __getitem__
    indexer = self.columns.get_loc(key)
  File "/scratch/ref/gdlab/metagenome-atlas_db/2.6a1/conda_envs/85ea1ba518acc8559abd39ebd8927742/lib/python3.6/site-packages/pandas/core/indexes/base.py", line 2900, in get_loc
    raise KeyError(key) from err
KeyError: 'reference'
```


Any idea?

**Comment by MrOlm** [comment]
I see- I think the error is resulting from trying to use centrality without doing secondary clustering. Either setting `-centW 0` or running secondary clustering should fix this problem

-Matt

**Comment by SilasK** [comment]
We test with secondary clustering

**Comment by brooksomics** [comment]
I ran into the same `KeyError: 'reference'` error today and `-centW 0` resolved my issue (dRep v3.2.0). 

Command:
```
dRep dereplicate drep_results \
        -g ./genomes/*.fasta \
        -p $threads \
        -pa $primarythresh \
        --SkipSecondary \
        --ignoreGenomeQuality \
        -centW 0
```

**Comment by MrOlm** [comment]
This bug is fixed in version 3.2.1 (it will automatically set centW to 0 if `--SkipSecondary` is on). Thanks for reporting!
